src/python/pants/backend/codegen/protobuf/python/python_protobuf_subsystem.py

```diff
@@ -81,10 +110,45 @@ class InjectPythonProtobufDependencies(InjectDependenciesRequest):
 
 @rule
 async def inject_dependencies(
-    _: InjectPythonProtobufDependencies, python_protobuf: PythonProtobufSubsystem
+    request: InjectPythonProtobufDependencies,
+    python_protobuf: PythonProtobufSubsystem,
+    # TODO(#12946): Make this a lazy Get once possible.
+    module_mapping: ThirdPartyPythonModuleMapping,
 ) -> InjectedDependencies:
-    addresses = await Get(Addresses, UnparsedAddressInputs, python_protobuf.runtime_dependencies)
-    return InjectedDependencies(addresses)
+    if python_protobuf.runtime_dependencies.values:
+        addresses = await Get(
+            Addresses, UnparsedAddressInputs, python_protobuf.runtime_dependencies
+        )
+        return InjectedDependencies(addresses)
+
+    if not python_protobuf.infer_runtime_dependency:
+        return InjectedDependencies()
+
+    result = [
+        find_python_runtime_library_or_error(
```

Comment:
Does this raise the error? Maybe better named `find_python_runtime_library_or_raise`? As I read it, I thought that `_or_error` could mean that it returns an error.